<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>FlashMate - Firmware Extraction Tool</string>
  </property>
  <property name="styleSheet">
   <string>QMainWindow {
    background-color: #f8fafc;
}
QGroupBox {
    border: 1px solid #e2e8f0;
    border-radius: 8px;
    margin-top: 6px;
    padding-top: 6px;
    font-weight: 600;
    color: #334155;
}
QGroupBox::title {
    subcontrol-origin: margin;
    left: 8px;
    padding: 0 6px 0 6px;
    background-color: #f8fafc;
}
QPushButton {
    background-color: #3b82f6;
    color: white;
    border: none;
    border-radius: 6px;
    padding: 8px 16px;
    font-weight: 500;
}
QPushButton:hover {
    background-color: #2563eb;
}
QPushButton:pressed {
    background-color: #1d4ed8;
}
QPushButton:disabled {
    background-color: #cbd5e1;
    color: #64748b;
}
QLineEdit {
    border: 1px solid #cbd5e1;
    border-radius: 6px;
    padding: 8px;
    background-color: white;
}
QLineEdit:focus {
    border-color: #3b82f6;
}
QComboBox {
    border: 1px solid #cbd5e1;
    border-radius: 6px;
    padding: 8px;
    background-color: white;
}
QComboBox:focus {
    border-color: #3b82f6;
}
QComboBox::drop-down {
    border: none;
}
QComboBox::down-arrow {
    image: none;
    border-left: 5px solid transparent;
    border-right: 5px solid transparent;
    border-top: 5px solid #64748b;
}
QTextEdit {
    border: 1px solid #cbd5e1;
    border-radius: 6px;
    background-color: white;
}
QProgressBar {
    border: 1px solid #cbd5e1;
    border-radius: 6px;
    text-align: center;
    background-color: #f1f5f9;
}
QProgressBar::chunk {
    background-color: #3b82f6;
    border-radius: 4px;
}
QTableWidget {
    border: 1px solid #cbd5e1;
    border-radius: 6px;
    background-color: white;
    gridline-color: #f1f5f9;
}
QTableWidget::item {
    padding: 8px;
}
QTableWidget::item:selected {
    background-color: #dbeafe;
    color: #334155;
}
QLabel {
    color: #475569;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="mainLayout">
    <property name="spacing">
     <number>15</number>
    </property>
    <property name="leftMargin">
     <number>20</number>
    </property>
    <property name="topMargin">
     <number>20</number>
    </property>
    <property name="rightMargin">
     <number>20</number>
    </property>
    <property name="bottomMargin">
     <number>20</number>
    </property>
    <item>
     <widget class="QGroupBox" name="inputGroup">
      <property name="title">
       <string>Input</string>
      </property>
      <layout class="QHBoxLayout" name="inputLayout">
       <property name="spacing">
        <number>20</number>
       </property>
       <item>
        <layout class="QVBoxLayout" name="fileSection">
         <property name="spacing">
          <number>12</number>
         </property>
         <item>
          <widget class="QLabel" name="file1Label">
           <property name="text">
            <string>File 1</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="file1Path">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="placeholderText">
            <string>Click or drag ZIP file here...</string>
           </property>
           <property name="styleSheet">
            <string>background-color: #f8fafc; border: 2px dashed #cbd5e1; cursor: pointer;</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="file2Label">
           <property name="text">
            <string>File 2</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="file2Path">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="placeholderText">
            <string>Click or drag ZIP file here...</string>
           </property>
           <property name="styleSheet">
            <string>background-color: #f8fafc; border: 2px dashed #cbd5e1; cursor: pointer;</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="variantSection">
         <property name="spacing">
          <number>12</number>
         </property>
         <item>
          <widget class="QLabel" name="variantLabel">
           <property name="text">
            <string>Car Variant</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="carVariantCombo">
           <property name="placeholderText">
            <string>Select variant...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="controlSection">
      <property name="spacing">
       <number>15</number>
      </property>

      <item>
       <widget class="QPushButton" name="processButton">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>50</height>
         </size>
        </property>
        <property name="text">
         <string>Start Processing</string>
        </property>
        <property name="styleSheet">
         <string>font-size: 14px; background-color: #3b82f6;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QProgressBar" name="progressBar">
        <property name="textVisible">
         <bool>true</bool>
        </property>
        <property name="format">
         <string>%p%</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="logGroup">
      <property name="title">
       <string>Progress &amp; Log</string>
      </property>
      <layout class="QVBoxLayout" name="logLayout">
       <property name="spacing">
        <number>12</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="progressLayout">
         <item>
          <widget class="QProgressBar" name="progressBar">
           <property name="textVisible">
            <bool>true</bool>
           </property>
           <property name="format">
            <string>%p%</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="openOutputFolder">
           <property name="text">
            <string>Open Output</string>
           </property>
           <property name="styleSheet">
            <string>background-color: #3b82f6;</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTextEdit" name="logText">
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="lineWrapMode">
          <enum>QTextEdit::WidgetWidth</enum>
         </property>
         <property name="styleSheet">
          <string>font-family: Consolas, Monaco, monospace; font-size: 11px; background-color: #f8f9fa;</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="outputPathLayout">
         <item>
          <widget class="QLabel" name="outputPathLabel">
           <property name="text">
            <string>Output Path:</string>
           </property>
           <property name="styleSheet">
            <string>font-weight: bold;</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="outputPath">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="styleSheet">
            <string>background-color: #fafafa; font-family: monospace;</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1000</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
